### DISCUSSION TITLE
Expected structured log format in Kubernetes pod logs

### DISCUSSION THREAD

**Discussion by nmehlei** [SigNoz/signoz#6201] [Q&A]
Hey all,
We are currently experimenting with SigNoz, but are struggling right now with getting structured logs to work in a Kubernetes environment.

Our (.NET) applications log to console via Serilog, which the OTLP agent forwards to SigNoz. As long as these are just text log lines everything is fine, yet when forwarding structured logs (e.g. with [Serilog's JSON formatters](https://github.com/serilog/serilog-formatting-compact)), the JSON seems to just be recognized as a string and therefore not actually being used:
<img width="1008" alt="SCR-20241016-mtbg" src="https://github.com/user-attachments/assets/9d44aff7-615e-482c-8cd0-a52356727727">

Which format do we need to use for our structured logs for it to be recognized? Alternatively, do we need to configure the OTel agent differently in our Kubernetes cluster? Our custom applications will log via structured logging, yet other applications that we're using need to continue pushing plain unstructured log files.

**Comment by srikanthccv** [comment]
Use logs pipelines to parse and structure them https://signoz.io/docs/logs-pipelines/guides/json/.
